flang/lib/Lower/OpenMP/ClauseProcessor.cpp

```diff
@@ -1003,6 +1006,20 @@ void ClauseProcessor::processMapObjects(
       }
     }
 
+    if (!mapperIdName.empty()) {
```

Comment:
Nit: If you do this, you don't have to call `mapperIdName.clear()` below, since the lifetime of the string copy is only what it needs to be.
```suggestion
    if (!mapperIdNameRef.empty()) {
      std::string mapperIdName = mapperIdNameRef.str();
```

Reply:
Nevermind, thank you for bringing to my attention that this is inside of a loop.